Source/WebCore/dom/Document.cpp

```diff
@@ -4163,6 +4163,20 @@ void Document::updateViewportArguments()
     }
 }
 
+void Document::metaElementColorSchemeChanged()
+{
+    RefPtr<CSSValue> protector = nullptr;
+    for (auto& metaElement : descendantsOfType<HTMLMetaElement>(rootNode())) {
+        const AtomString& nameValue = metaElement.attributeWithoutSynchronization(nameAttr);
+        if ((equalLettersIgnoringASCIICase(nameValue, "color-scheme"_s) || equalLettersIgnoringASCIICase(nameValue, "supported-color-schemes"_s)) && (protector = CSSParser::parseSingleValue(CSSPropertyColorScheme, metaElement.attributeWithoutSynchronization(contentAttr), CSSParserContext(document()))))
```

Comment:
I think we should create the `CSSParserContext` once, rather than every loop iteration.